love -> 0.10 added

leenaars opened the pull request:
LÖVE has different versions with partially incompatible API's. This commit adds the latest stable upstream version. I don't know what the canonical way of dealing with this is, so I left the oldest version (0.8) intact as "default". 0.9 has been added before by someone else in a similar way..

7c6f434c commented:
So you add 0.8.nix but never use it?

jgillich commented:
Please use `fetchFromBitbucket`.

jagajaga commented on pkgs/development/interpreters/love/0.10.nix:
```diff
@@ -0,0 +1,37 @@
+    homepage = "http://love2d.org";
+    description = "A Lua-based 2D game engine/scripting language";
+    license = stdenv.lib.licenses.zlib;
+
```
Please remove this empty line.

jagajaga commented:
We do already have https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/interpreters/love.

jagajaga commented:
So you can just update the default version to `0.10`. I don't know why do we need all of the versions.

leenaars committed:
love: 0.8 -> 0.10.1

leenaars commented:
Meanwhile (5 hours ago) a new version 0.10.1 was released. This was added to default.nix instead of 0.10.0. The games created with this lib are kind of specific for that version, and some interesting games don't follow the API upgrades so having multiple versions of LOVE around makes sense. Can't have too much LOVE anyway ;)

Used fetchFromBitbucket instead of direct fetchurl.

jagajaga commented:
Ok, let's wait for travis's result :)

leenaars commented:
Not only was there a new point release, but this also involved a new automagic based toolchain to preconfigure. Added this. The different versions allow people building packages with these different versions to pin to a specific version, as their API's are different and not every game will be updated.

jagajaga commented:
You don't need to define `0.10` as a separate file if it is in `default.nix`. So please remove `default.nix` and in `all-packages` do
```
love_0_8 = callPackage ../development/interpreters/love/0.8.nix { };
love_0_10 = callPackage ../development/interpreters/love/0.10.nix { };
love = love_0_10;
```

jagajaga commented on pkgs/top-level/all-packages.nix:
```diff
@@ -5221,9 +5221,11 @@ let
+  love_0_8 = callPackage ../development/interpreters/love/0.8.nix { };
   love_0_9 = callPackage ../development/interpreters/love/0.9.nix { };
+  love_0_10 = callPackage ../development/interpreters/love/0.10.nix { };
+  love = love_0_10 
```
Missing `;`.

jagajaga commented:
Don't forget to delete `default.nix`.

leenaars commented:
@jagajaga: removed it, I thought is was "mandatory" to always have a default.nix in a folder.
I by the way also removed love_luajit entry, after grepping through nixpkgs and not finding a single mention. Especially since all current versions *are* using luajit already...
